### PR TITLE
Add API reference for locale keys - merge after new ci Run!

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### Description (required)

Add API reference for locale keys.

#### For Astro 4.0

### CODE REVIEW

File: src/content/docs/en/reference/api-reference.mdx [hunk 1]
```diff
@@ -452,6 +451,18 @@ const orders = Array.from(Astro.locals.orders.entries());
 </ul>
 ```
 
+### `Astro.preferredLocale`
+
+`Astro.preferredLocale` represents the preferred locale of the user. It's computed by checking the supported locales in your `i18n.locales` array and locales supported by the users's browser via the header `Accept-Language`. This value is `undefined` if no such match exists. This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
+
+### `Astro.preferredLocaleList`
+
+`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your locales array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](https://docs.astro.build/en/reference/configuration-reference/#experimentali18nlocales). This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:460]:
```suggestion
`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your locales array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](/en/reference/configuration-reference/#i18nlocales). This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```

Comment (TheOtterlord) [reply src/content/docs/en/reference/api-reference.mdx:460]: oops, good catch :sweat_smile:

Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx]:
```suggestion
`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your locales array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](en/reference/configuration-reference/#i18nlocales). This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 2]
```diff
@@ -452,6 +451,18 @@ const orders = Array.from(Astro.locals.orders.entries());
 </ul>
 ```
 
+### `Astro.preferredLocale`
+
+`Astro.preferredLocale` represents the preferred locale of the user. It's computed by checking the supported locales in your `i18n.locales` array and locales supported by the users's browser via the header `Accept-Language`. This value is `undefined` if no such match exists. This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:456]:
```suggestion
`Astro.preferredLocale` is a computed value that represents the preferred locale of the user. 

It is computed by checking the configured locales in your `i18n.locales` array and locales supported by the users's browser via the header `Accept-Language`. This value is `undefined` if no such match exists.

This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 3]
```diff
@@ -452,6 +451,18 @@ const orders = Array.from(Astro.locals.orders.entries());
...
+
+### `Astro.preferredLocaleList`
+
+`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your locales array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](/en/reference/configuration-reference/#i18nlocales). This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:460]:
```suggestion
`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. 

If none of the browser's requested languages are found in your locales array, then the value is `[]`: you do not support any of your visitor's preferred locales.

If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](/en/reference/configuration-reference/#i18nlocales): all of your supported locales will be considered equally preferred by a visitor with no preferences. 

This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
```

**Review by sarah11918 — APPROVED** [review]
This is amazingly helpful, thank you for putting this together, Otterlord! I just made some light formatting edits for clarity!  🙌

**Pull request merged**